### ISSUE TITLE
Infinite loss during training

### ISSUE THREAD

**Issue by Bizilizi** [stoneMo/DeepAVFusion#5]
Hi my name is Daniil, 

I've been trying to reproduce your work for my research project. However, I stumbled upon a problem that I can solve for the last couple of days. 

I'm fine-tuning your model on the VGGSound dataset with the following provided config:

```bash
...
model.fusion.attn_ratio=0.25 \
model.fusion.mlp_ratio=1.0 \
data.dataset=vggsound \
opt.epochs=60 \
opt.warmup_epochs=10 \
opt.batch_size=64 \
opt.accum_iter=4 \
opt.blr=0.3
...
```

The loss behaves very strangely (goes up) and, at some point, becomes infinite. 
```logs
[18:29:30.107704] => saved checkpoint '/home/wiss/zverev/DeepAVFusion/checkpoints/deepavfusion_vitb_vggsound_ep200/finetunee_vggsound_1gpu/checkpoints/checkpoint_latest.pth' (epoch 6)
[18:29:35.540031] [Train][Ep-6/60]  [   0/2870]  eta: 4:19:22  lr: 0.1800 (0.1800)  time: 5.4226  data: 4.4980  max mem: 22233
[18:31:10.891985] [Train][Ep-6/60]  [ 100/2870]  eta: 0:46:03  lr: 0.1806 (0.1805)  loss: 2602.3862 (2591.0107)  grad_norm: 105.1632 (105.8583)  amp_scale: 1.0000 (1.0000)  time: 0.9583  data: 0.0002  max mem: 22233
[18:32:46.756207] [Train][Ep-6/60]  [ 200/2870]  eta: 0:43:31  lr: 0.1817 (0.1810)  loss: 2591.0964 (2593.0216)  grad_norm: 109.0842 (110.6869)  amp_scale: 1.0000 (1.0000)  time: 0.9594  data: 0.0002  max mem: 22233
[18:34:22.767134] [Train][Ep-6/60]  [ 300/2870]  eta: 0:41:38  lr: 0.1827 (0.1816)  loss: 2591.1013 (2593.0779)  grad_norm: 112.7342 (144.8024)  amp_scale: 1.0000 (1.0000)  time: 0.9600  data: 0.0002  max mem: 22233
[18:35:58.689299] [Train][Ep-6/60]  [ 400/2870]  eta: 0:39:53  lr: 0.1838 (0.1821)  loss: 2597.5073 (2594.1809)  grad_norm: 110.1128 (135.9338)  amp_scale: 1.0000 (1.0000)  time: 0.9592  data: 0.0002  max mem: 22233
[18:37:34.559442] [Train][Ep-6/60]  [ 500/2870]  eta: 0:38:11  lr: 0.1848 (0.1826)  loss: 2620.9985 (2598.3037)  grad_norm: 123.8003 (136.7234)  amp_scale: 1.0000 (1.0000)  time: 0.9586  data: 0.0002  max mem: 22233
[18:39:10.562182] [Train][Ep-6/60]  [ 600/2870]  eta: 0:36:32  lr: 0.1859 (0.1831)  loss: 2587.2302 (2597.5696)  grad_norm: 111.5621 (135.9826)  amp_scale: 1.0000 (1.0000)  time: 0.9589  data: 0.0002  max mem: 22233
[18:40:46.727059] [Train][Ep-6/60]  [ 700/2870]  eta: 0:34:54  lr: 0.1869 (0.1837)  loss: 2584.9600 (2596.0281)  grad_norm: 119.8937 (184.6462)  amp_scale: 1.0000 (1.0000)  time: 0.9689  data: 0.0002  max mem: 22233
[18:42:22.596981] [Train][Ep-6/60]  [ 800/2870]  eta: 0:33:16  lr: 0.1879 (0.1842)  loss: 2621.0837 (2600.1229)  grad_norm: 23706.7461 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9583  data: 0.0002  max mem: 22233
[18:43:58.546084] [Train][Ep-6/60]  [ 900/2870]  eta: 0:31:38  lr: 0.1890 (0.1847)  loss: 2661.7180 (2607.0020)  grad_norm: 209098.3750 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9582  data: 0.0002  max mem: 22233
[18:45:34.344497] [Train][Ep-6/60]  [1000/2870]  eta: 0:30:00  lr: 0.1900 (0.1852)  loss: 2633.8225 (2610.9554)  grad_norm: 33364.2344 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9581  data: 0.0002  max mem: 22233
[18:47:10.144154] [Train][Ep-6/60]  [1100/2870]  eta: 0:28:23  lr: 0.1911 (0.1857)  loss: 2628.5688 (2611.7988)  grad_norm: 1844.7959 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9572  data: 0.0002  max mem: 22233
[18:48:45.932740] [Train][Ep-6/60]  [1200/2870]  eta: 0:26:46  lr: 0.1921 (0.1863)  loss: 2634.6538 (2614.0705)  grad_norm: 1394.8171 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9578  data: 0.0002  max mem: 22233
[18:50:21.760627] [Train][Ep-6/60]  [1300/2870]  eta: 0:25:10  lr: 0.1932 (0.1868)  loss: 2638.2239 (2616.6169)  grad_norm: 388.1147 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9584  data: 0.0002  max mem: 22233
[18:51:57.558045] [Train][Ep-6/60]  [1400/2870]  eta: 0:23:33  lr: 0.1942 (0.1873)  loss: 2630.1072 (2617.6781)  grad_norm: 334.3351 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9574  data: 0.0002  max mem: 22233
[18:53:33.336535] [Train][Ep-6/60]  [1500/2870]  eta: 0:21:57  lr: 0.1953 (0.1878)  loss: 2612.7175 (2618.2016)  grad_norm: 215.3917 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9581  data: 0.0002  max mem: 22233
[18:55:09.144747] [Train][Ep-6/60]  [1600/2870]  eta: 0:20:20  lr: 0.1963 (0.1884)  loss: 2626.8494 (2618.5154)  grad_norm: 231.8829 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9582  data: 0.0002  max mem: 22233
[18:56:44.968860] [Train][Ep-6/60]  [1700/2870]  eta: 0:18:44  lr: 0.1974 (0.1889)  loss: 2625.1921 (2619.3574)  grad_norm: 207.9724 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9583  data: 0.0002  max mem: 22233
[18:58:20.763395] [Train][Ep-6/60]  [1800/2870]  eta: 0:17:08  lr: 0.1984 (0.1894)  loss: 2624.8418 (2620.1069)  grad_norm: 195.9008 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9583  data: 0.0002  max mem: 22233
[18:59:56.578024] [Train][Ep-6/60]  [1900/2870]  eta: 0:15:31  lr: 0.1994 (0.1899)  loss: 2634.6790 (2620.7790)  grad_norm: 204.9135 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9580  data: 0.0002  max mem: 22233
[19:01:32.417646] [Train][Ep-6/60]  [2000/2870]  eta: 0:13:55  lr: 0.2005 (0.1905)  loss: 2643.8906 (2621.9943)  grad_norm: 203.2748 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9578  data: 0.0002  max mem: 22233
[19:03:08.420083] [Train][Ep-6/60]  [2100/2870]  eta: 0:12:19  lr: 0.2015 (0.1910)  loss: 2640.2014 (2622.9284)  grad_norm: 192.7788 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9588  data: 0.0002  max mem: 22233
[19:04:44.298179] [Train][Ep-6/60]  [2200/2870]  eta: 0:10:43  lr: 0.2026 (0.1915)  loss: 2655.4219 (2624.3071)  grad_norm: 208.8167 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9585  data: 0.0002  max mem: 22233
[19:06:20.238466] [Train][Ep-6/60]  [2300/2870]  eta: 0:09:07  lr: 0.2036 (0.1920)  loss: 2628.4998 (2625.0446)  grad_norm: 220.0584 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9595  data: 0.0002  max mem: 22233
[19:07:56.080570] [Train][Ep-6/60]  [2400/2870]  eta: 0:07:31  lr: 0.2047 (0.1925)  loss: 2651.0522 (2626.4267)  grad_norm: 223.6025 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9585  data: 0.0002  max mem: 22233
[19:09:31.958372] [Train][Ep-6/60]  [2500/2870]  eta: 0:05:55  lr: 0.2057 (0.1931)  loss: 2658.3738 (2628.1080)  grad_norm: 228.0880 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9588  data: 0.0002  max mem: 22233
[19:11:07.860534] [Train][Ep-6/60]  [2600/2870]  eta: 0:04:19  lr: 0.2068 (0.1936)  loss: 2669.9233 (2629.9844)  grad_norm: 243.9257 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9589  data: 0.0002  max mem: 22233
[19:12:43.835691] [Train][Ep-6/60]  [2700/2870]  eta: 0:02:43  lr: 0.2078 (0.1941)  loss: 2663.7888 (2631.5457)  grad_norm: 268.7336 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9595  data: 0.0002  max mem: 22233
[19:14:19.992423] [Train][Ep-6/60]  [2800/2870]  eta: 0:01:07  lr: 0.2089 (0.1946)  loss: 2668.5396 (2633.1116)  grad_norm: 297.8978 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9586  data: 0.0002  max mem: 22233
[19:15:26.213511] [Train][Ep-6/60]  [2869/2870]  eta: 0:00:00  lr: 0.2096 (0.1950)  loss: 2696.6846 (2636.6151)  grad_norm: 312.9929 (inf)  amp_scale: 1.0000 (1.0000)  time: 0.9598  data: 0.0004  max mem: 22233
[19:15:26.214938] [Train][Ep-6/60] Total time: 0:45:56 (0.9603 s / it)
[19:15:26.215399] Syncing meters...
[19:15:26.216698] Averaged stats: lr: 0.2096 (0.1950)  loss: 2696.6846 (2636.6151)  grad_norm: 312.9929 (inf)  amp_scale: 1.0000 (1.0000)
```

It seems like you already encountered this problem before as you have this fallback (incorrect Python syntax, btw [link](https://github.com/stoneMo/DeepAVFusion/blob/2768e7f77b65c45e39e019b0b13315f0b4c4fddd/eval_finetune.py#L299)):
```python
if not math.isfinite(loss.item()):
    raise f"Loss is {loss.item()}, stopping training"
```

Any ideas on how to solve it? 
Alternatively, could you please share checkpoints on the VGGSound fine-tuned version of your model?

**Comment by Bizilizi** [comment]
In addition, the linprob scenario works well; however, I struggle to reproduce your results. 

The last checkpoint with the provided training parameters gives the following results:

**Audio accuracy:** 0.11
**Visual accuracy:** 0.07
**Fusion accuracy:** 0.36

**Comment by Bizilizi** [comment]
In addition, the linear probe setup works well, but I struggle to reproduce your results with the provided hyperparameters.

**Audio accuracy:** 0.11
**Visual accuracy:** 0.07
**Fusion accuracy:** 0.36

I run the following command:
```bash
# libprobe
PYTHONPATH=. python launcher.py \
    --config-name=linprobe \
    job_name=linprobe_vggsound_4gpu \
    pretrain_job_name=deepavfusion_vitb_vggsound_ep200 \
    model.fusion.attn_ratio=0.25 \
    model.fusion.mlp_ratio=1.0 \
    data.dataset=vggsound \
    data.data_path=/tmp/vggsound \
    opt.epochs=60 \
    opt.warmup_epochs=10 \
    opt.batch_size=64 \
    opt.accum_iter=4 \
    opt.blr=0.3 \
    env.ngpu=4 \
    env.world_size=1    
```

**Comment by stoneMo** [comment]
Hi, thanks for your interest in our work! Please follow the same hyper-parameter settings in the paper.
